crates/matrix-sdk/src/send_queue.rs

```diff
@@ -1145,24 +1145,41 @@ impl SendHandle {
 /// meaningful, i.e. the ones that wouldn't be overridden after applying the
 /// others.
 fn canonicalize_dependent_events(dependent: &[DependentQueuedEvent]) -> Vec<DependentQueuedEvent> {
```

Comment:
So I think previously we assumed everything in `dependent` would be for the same parent and now we don't? Is this fixing an existing bug?

Reply:
Yes indeed; this function was already handling all the dependent events for all the queued events, so this was a bug. The new test was failing on main.

Reply:
I assume there was not a github issue for it though? It feels weird to fix a real thing and have no audit trail, but I guess creating a whole issue just to close it seems a bit heavy.

Reply:
No github issue, I've found out about the issues while addressing review comments in the PR that introduced the feature, so I figured I could address those in a followup, instead of sneaking in the changes in the approved PR.